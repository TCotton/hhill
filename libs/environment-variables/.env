NEXT_PUBLIC_RELEASE_UNIT_TESTING_TWO='hello'
