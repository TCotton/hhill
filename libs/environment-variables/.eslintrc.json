{
  "extends": ["../../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "rules": {
        "dot-notation": "off"
      }
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
