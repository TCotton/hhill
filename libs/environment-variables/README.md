# environment-variables

This library was generated with [Nx](https://nx.dev).

## Building

Run `nx build environment-variables` to build the library.

## Running unit tests

Run `nx test environment-variables` to execute the unit tests via [Jest](https://jestjs.io).
