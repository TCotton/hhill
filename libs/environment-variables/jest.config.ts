/* eslint-disable */
export default {
  displayName: 'environment-variables',
  preset: '../../jest.preset.js',
  globals: {
    'ts-jest': {
      tsconfig: '<rootDir>/tsconfig.spec.json',
    },
  },
  transform: {
    '^(?!.*\\.(js|jsx|ts|tsx|css|json)$)': '@nrwl/react/plugins/jest',
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nrwl/next/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'd.ts'],
  coverageDirectory: '../../coverage/libs/environment-variables',
};
