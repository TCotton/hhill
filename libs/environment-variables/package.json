{
  "name": "@hhill/environment-variables",
  "version": "0.0.1",
  "type": "commonjs"
}
