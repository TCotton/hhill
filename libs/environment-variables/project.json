{
  "name": "environment-variables",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/environment-variables/src",
  "targets": {
    "build": {
      "executor": "@nrwl/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/environment-variables",
        "main": "libs/environment-variables/src/index.ts",
        "tsConfig": "libs/environment-variables/tsconfig.lib.json",
        "assets": ["libs/environment-variables/*.md"]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/environment-variables/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/environment-variables"],
      "options": {
        "jestConfig": "libs/environment-variables/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
