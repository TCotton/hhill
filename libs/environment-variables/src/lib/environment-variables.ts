export function environmentVariables(): string {
  return 'environment-variables';
}
