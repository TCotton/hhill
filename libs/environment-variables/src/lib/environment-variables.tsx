import React from 'react'
import * as path from 'path'
import * as dotenv from 'dotenv'
dotenv.config({ path: path.join(__dirname, '../../../../.env.local') })
const featureList: { [x: string]: string } = {
  unitTesting: process.env['NEXT_PUBLIC_RELEASE_UNIT_TESTING'] || '',
  unitTestingTwo: process.env['NEXT_PUBLIC_RELEASE_UNIT_TESTING_TWO'] || ''
}

const stringToBoolean = (string: string) => {
  switch (string) {
    case 'false':
      return false
    case 'true':
      return true
    default:
      return false
  }
}

const featureToggle = {
  unitTesting: (): boolean => {
    return stringToBoolean(featureList['unitTesting'])
  },
  unitTestingTwo: (): boolean => {
    return stringToBoolean(featureList['unitTestingTwo'])
  }
}

const toggle = (name: string) => {
  return featureList[name]
}

const toggleOn =
  (featureName: string, ComposedComponent: React.FunctionComponent) =>
  (props: JSX.IntrinsicAttributes & { children?: React.ReactNode }) => {
    return stringToBoolean(toggle(featureName)) ? (
      <ComposedComponent {...props} />
    ) : null
  }

export { featureToggle, featureList, toggleOn }
