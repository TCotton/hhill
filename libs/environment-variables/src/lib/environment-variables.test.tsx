import * as React from 'react'
import { FC } from 'react'
import { featureToggle, toggleOn } from './environment-variables'
import { render, screen, cleanup } from '@testing-library/react'

interface NavProps {
  greetings: string
}

describe('feature toggle', () => {
  const Nav: FC<NavProps> = ({ greetings = 'Hello' }) => {
    return (
      <div data-testid="element">
        {greetings}
        {featureToggle.unitTesting() && (
          <span data-testid="test">unit testing</span>
        )}
        {featureToggle.unitTestingTwo() && (
          <span data-testid="testtwo">unit testing</span>
        )}
      </div>
    )
  }

  afterEach(cleanup)

  describe('featureToggle - undefined && true for test env variable', () => {
    it(`should return undefined || 'true' || 'false'`, () => {
      expect(process.env.NEXT_PUBLIC_RELEASE_UNIT_TESTING).toBe(undefined)
      expect(process.env.NEXT_PUBLIC_RELEASE_UNIT_TESTING_TWO).toBe('false')
    })
    it('should return false if unitTesting called', () => {
      expect(featureToggle.unitTesting()).toBe(false)
    })
    it('should false false if unitTestingTwo is called', () => {
      expect(featureToggle.unitTestingTwo()).toBe(false)
    })
    xit('should render with correct elements', () => {
      const { baseElement } = render(<Nav greetings="" />)
      expect(baseElement).toMatchSnapshot()
      expect(screen.queryByTestId('payment')).toBeInTheDocument()
      expect(screen.queryByTestId('subscription')).toBeInTheDocument()
      expect(screen.queryByTestId('test')).not.toBeInTheDocument()
      expect(screen.queryByTestId('testtwo')).not.toBeInTheDocument()
      expect(screen.queryByTestId('reviews')).toBeInTheDocument()
      expect(screen.queryByTestId('productRecommendations')).toBeInTheDocument()
    })
    it(`should NOT render component if feature toggle is undefined`, () => {
      const Component = toggleOn('unitTesting', Nav)
      const { container, baseElement } = render(<Component />)
      expect(baseElement).toMatchSnapshot()
      expect(container.firstChild).toBe(null)
    })
    it(`should NOT render component if feature toggle is 'false'`, () => {
      const Component = toggleOn('unitTestingTwo', Nav)
      const { container, baseElement } = render(<Component />)
      expect(baseElement).toMatchSnapshot()
      expect(container.firstChild).toBe(null)
    })
  })
})
