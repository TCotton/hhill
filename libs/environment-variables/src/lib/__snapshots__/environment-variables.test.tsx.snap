// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feature toggle featureToggle - undefined && true for test env variable should NOT render component if feature toggle is 'false' 1`] = `
<body>
  <div />
</body>
`;

exports[`feature toggle featureToggle - undefined && true for test env variable should NOT render component if feature toggle is undefined 1`] = `
<body>
  <div />
</body>
`;
